### PR TITLE
refactor: rm boost

### PR DESCRIPTION
**Pull request by emo-coder**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

### CODE REVIEW

File: src/sdk/sql_sdk_base_test.cc [hunk 1]
```diff
@@ -112,12 +114,12 @@ void SQLSDKTest::CreateProcedure(hybridse::sqlcase::SqlCase& sql_case,  // NOLIN
...
+        absl::StrReplaceAll(sql, {{placeholder, sql_case.inputs()[i].name_}});
     }
-    boost::replace_all(
-        sql, "{auto}",
-        hybridse::sqlcase::SqlCase::GenRand("auto_t") + std::to_string(static_cast<int64_t>(time(NULL))));
-    boost::trim(sql);
+    absl::StrReplaceAll(
+        sql, {{"{auto}",
+        hybridse::sqlcase::SqlCase::GenRand("auto_t") + std::to_string(static_cast<int64_t>(time(NULL)))}});
+    absl::StrJoin(absl::StrSplit(sql, " "), "");
```
Comment (dl239) [inline src/sdk/sql_sdk_base_test.cc]: StripAsciiWhitespace or  RemoveExtraAsciiWhitespace

File: src/sdk/sql_sdk_base_test.cc [hunk 2]
```diff
@@ -70,7 +72,7 @@ void SQLSDKTest::CreateTables(hybridse::sqlcase::SqlCase& sql_case,  // NOLINT
         std::string create;
         if (sql_case.BuildCreateSqlFromInput(i, &create, partition_num) && !create.empty()) {
             std::string placeholder = "{" + std::to_string(i) + "}";
-            boost::replace_all(create, placeholder, sql_case.inputs()[i].name_);
+            absl::StrReplaceAll(create, {{placeholder, sql_case.inputs()[i].name_}});
```
Comment (dl239) [inline src/sdk/sql_sdk_base_test.cc]: create = absl::StrReplaceAll(create, {{placeholder, sql_case.inputs()[i].name_}});

File: src/sdk/file_option_parser.h [hunk 3]
```diff
@@ -44,7 +45,7 @@ class FileOptionsParser {
     ::openmldb::base::Status Parse(const std::shared_ptr<hybridse::node::OptionsMap>& options_map) {
         for (const auto& item : *options_map) {
             std::string key = item.first;
-            boost::to_lower(key);
+            absl::AsciiStrToLower(key);
```
Comment (dl239) [inline src/sdk/file_option_parser.h]: absl::AsciiStrToLower(&key);

File: src/sdk/mini_cluster_batch_bm.cc [hunk 4]
```diff
@@ -705,7 +705,7 @@ last join {2} order by {2}.x7 on {0}.c2 = {2}.x2 and {0}.c7 - {ts_diff} >= {2}.x
 last join {3} order by {3}.x7 on {0}.c3 = {3}.x3 and {0}.c7 - {ts_diff} >= {3}.x7
 last join {4} order by {4}.x7 on {0}.c4 = {4}.x4 and {0}.c7 - {ts_diff} >= {4}.x7;
 )";
-    boost::replace_all(sql_case.sql_str_, "{ts_diff}", std::to_string(state.range(0) * 1000 / 2));
+    absl::StrReplaceAll({{"{ts_diff}", std::to_string(state.range(0) * 1000 / 2)}}, sql_case.sql_str_);
```
Comment (dl239) [inline src/sdk/mini_cluster_batch_bm.cc]: &

**Review by dl239 — APPROVED** [review]

**Review by vagetablechicken — APPROVED** [review]

**Pull request merged**